cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/k8s/PreviewRunnerMain.java

```diff
@@ -17,47 +17,137 @@
+    return Arrays.asList(
+      new PreviewRunnerManagerModule().getDistributedModules(),
+      new DataSetServiceModules().getStandaloneModules(),
+      new DataSetsModules().getStandaloneModules(),
+      new AppFabricServiceRuntimeModule().getStandaloneModules(),
+      new ProgramRunnerRuntimeModule().getStandaloneModules(),
+      new MetricsStoreModule(),
+      new MessagingClientModule(),
+      new AuditModule(),
+      new SecureStoreClientModule(),
+      new MetadataReaderWriterModules().getStandaloneModules(),
+      getDataFabricModule(),
+      new DFSLocationModule(),
+      new MetadataServiceModule(),
+      new AuthorizationModule(),
+      new AuthorizationEnforcementModule().getMasterModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(ExploreClient.class).to(UnsupportedExploreClient.class);
+          bind(PreviewRequestPollerInfoProvider.class).toInstance(
+            () -> Bytes.toBytes(GSON.toJson(previewRequestPollerInfo)));
+        }
+      }
+    );
   }
 
   @Override
   protected void addServices(Injector injector, List<? super Service> services,
                              List<? super AutoCloseable> closeableResources, MasterEnvironment masterEnv,
                              MasterEnvironmentContext masterEnvContext, PreviewRunnerOptions options) {
+    Service previewRunnerManager = (Service) injector.getInstance(PreviewRunnerManager.class);
+    previewRunnerManager.addListener(new ServiceListenerAdapter() {
+      @Override
+      public void terminated(Service.State from) {
+        terminate();
+      }
+
+      @Override
+      public void failed(Service.State from, Throwable failure) {
+        terminate();
+      }
 
+      private void terminate() {
+        System.exit(0);
+      }
+    }, command -> {
+      Thread thread = new Thread(command, "PreviewRunnerMainTerminator");
+      thread.setDaemon(true);
+      thread.start();
+    });
+    services.add(previewRunnerManager);
   }
 
   @Nullable
   @Override
   protected LoggingContext getLoggingContext(PreviewRunnerOptions options) {
-    return null;
+    String instanceName = readFile(options.getInstanceNameFilePath());
+    String instanceUid = readFile(options.getInstanceUidFilePath());
+    LOG.info("Instance name: {}, Instance UID: {}", instanceName, instanceUid);
+    previewRequestPollerInfo = new PreviewRequestPollerInfo(instanceName, instanceUid);
+    LOG.info("Instance id: {}, Instance UID: {}", previewRequestPollerInfo.getInstanceId(),
+             previewRequestPollerInfo.getInstanceUid());
+    return new ServiceLoggingContext(NamespaceId.SYSTEM.getNamespace(),
+                                     Constants.Logging.COMPONENT_NAME,
+                                     Constants.Service.PREVIEW_HTTP);
+  }
+
+  private static String readFile(String fileName) {
+    try {
+      return new String(Files.readAllBytes(Paths.get(fileName)));
```

Comment:
Always provide a charset when turning bytes to string

Reply:
Using `Bytes.toString()` now to use UTF_8 encoding.